rest/handler.go

```diff
@@ -329,8 +329,8 @@ func (h *handler) checkAuth(context *db.DatabaseContext) (err error) {
 		if context.Options.UnsupportedOptions.OidcTestProvider.Enabled && strings.HasSuffix(h.rq.URL.Path, "/_oidc_testing/token") {
 			if username, password := h.getBasicAuth(); username != "" && password != "" {
 				provider := context.Options.OIDCOptions.Providers.GetProviderForIssuer(issuerUrlForDB(h, context.Name), testProviderAudiences)
-				if provider != nil && provider.ClientID != nil && provider.ValidationKey != nil {
-					if *provider.ClientID == username && *provider.ValidationKey == password {
+				if provider != nil && provider.ClientID != "" && provider.ValidationKey != nil {
```

Comment:
Looks like this was just guarding against nil pointer exceptions when comparing ClientID below.  I don't think there's any reason to check for empty string, is there?

Reply:
That's fair. We're already  checking provider.ClientID == username down below. Fixed in latest revision.